aten/src/ATen/native/Distance.cpp

```diff
@@ -14,9 +14,12 @@ Tensor pairwise_distance(const Tensor& x1, const Tensor& x2, double p, double ep
 }
 
 // This is to guarantee that the contiguous memory is passed to the backward pass
+// TODO This currently enforces that the entire array is contiguous, but the
+// batches don't really have to be for efficiency sake, meaning there may be a
+// better way to only force the last two dimensions as contiguous.
```

Comment:
It's not too hard to do a contiguity check for just the last two dims; you just need to check that stride[d-1] is 1, and stride[d-2] is size[d-1].

Reply:
Yeah, but then I'd also have to work on more general handling of the remaining dimensions. I think there's probably a more general way to write this code, I just wanted to make sure that some form of pdist is available for batched operations. Also, I thinks it's very unlikely for this whole tensor to not be contiguous, so this really shouldn't be that big of an issue.